From 0f2aebe277622d46f294e3e50ad8fea9121b8a1b Mon Sep 17 00:00:00 2001
From: Igal Liberman <igall@marvell.com>
Date: Tue, 8 Nov 2016 19:23:16 +0200
Subject: [PATCH 0570/1345] phy: comphy: cp110: xfi: make sure that 40 data
 bits mode is disabled

commit  097cdd93070ef4b9d3dd5d1fc69d0ac3c49eb6cf from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- 40 data bits mode should be enabled only for SATA
- 40 data bits mode isn't cleared by hard reset (HW issue)
- Need to make sure that this mode is disabled (can't rely on u-boot).

When support to all modes will be added, this can be done in power-off
execution.

Change-Id: Id6b74dc12f7e67e3ed38a00ca147fb1646556a7b
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/33625
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/phy/phy-mvebu-comphy.c |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/phy/phy-mvebu-comphy.c b/drivers/phy/phy-mvebu-comphy.c
index 9a85f18..8d8375d 100644
--- a/drivers/phy/phy-mvebu-comphy.c
+++ b/drivers/phy/phy-mvebu-comphy.c
@@ -418,6 +418,13 @@ static int mvebu_comphy_xfi_power_on(struct mvebu_comphy_priv *priv,
 	data |= 0x0 << COMMON_PHY_CFG1_PIPE_SELECT_OFFSET;
 	reg_set(comphy_addr + COMMON_PHY_CFG1_REG, data, mask);
 
+	/* Make sure that 40 data bits is disabled
+	 * This bit is not cleared by reset
+	 */
+	mask = COMMON_PHY_CFG6_IF_40_SEL_MASK;
+	data = 0 << COMMON_PHY_CFG6_IF_40_SEL_OFFSET;
+	reg_set(comphy_addr + COMMON_PHY_CFG6_REG, data, mask);
+
 	/* Select Baud Rate of Comphy And PD_PLL/Tx/Rx */
 	mask = SD_EXTERNAL_CONFIG0_SD_PU_PLL_MASK;
 	data = 0x0 << SD_EXTERNAL_CONFIG0_SD_PU_PLL_OFFSET;
-- 
1.7.9.5

